src/containers/AppConfiguration/AppConfiguration.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useState, useEffect, useRef } from "react";
+import ContentstackAppSdk from "@contentstack/app-sdk";
+import { IInstallationData } from "@contentstack/app-sdk/dist/src/types";
+import Icon from "../../assets/appconfig.svg";
+import localeTexts from "../../common/locales/en-us/index";
+import parse from "html-react-parser";
+import { merge } from "../../common/utils/functions";
+import "../../index.css";
+
+interface AppState {
+  installationData: IInstallationData;
+  setInstallationData: (event: any) => any;
+  appSdkInitialized: boolean;
+}
+
+const AppConfigurationExtension: React.FC = () => {
+  const usernameRef = useRef<any>("");
+  const secretRef = useRef<any>("");
+  const [config, setConfig] = useState<AppState>({
+    installationData: {
+      configuration: {
+        username: usernameRef,
+      },
+      serverConfiguration: {
+        secret: secretRef,
+      },
+    },
+    setInstallationData: (event): any => {
+      return;
+    },
+    appSdkInitialized: false,
+  });
+
+  useEffect(() => {
+    ContentstackAppSdk.init().then(async (appSdk) => {
```

Comment:
init is not needed again. We initialise it in Marketplace Provider. You can use useApplocationHook to get access to location object

Reply:
We also have useInstalltionData hook